public/app/core/components/DynamicImports/SafeDynamicImport.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { lazy, Suspense } from 'react';
+import { css } from 'emotion';
+import { ErrorBoundary } from '@grafana/ui';
+import { LoadingChunkPlaceHolder } from './LoadingChunkPlaceHolder';
+import { ShowLoadingChunkError } from './ShowLoadingChunkError';
+import { ShowErrorWithStack } from '@grafana/ui/src/components/ErrorBoundary/ShowErrorWithStack';
+
+function getAlertPageStyle(): string {
+  return css`
+    width: 508px;
+    margin: 128px auto;
+  `;
+}
+
+export const isChunkError = (error: Error): boolean => {
+  if (error && error.message && error.message.match(/Loading chunk \w+ failed/)) {
```

Comment:
I wonder if we could make this a bit more explicit. I think it should be possible to do:
```
lazy(() => {
  try {
    await importStatement
  } catch (error) {
    // rethrow with error type or some flag
  }
)
```
Because honestly not sure if the message itself cannot change between versions of browser or webpack or whatever is providing the function itself.

Reply:
Very good point! Will have a look